src/brs/db/sql/SqlBlockDb.java

```diff
@@ -102,9 +103,26 @@ public Block loadBlock(BlockRecord r) throws BurstException.ValidationException
     long nonce                      = r.getNonce();
     byte[] blockATs                 = r.getAts();
 
-    return new Block(version, timestamp, previousBlockId, totalAmountNQT, totalFeeNQT, payloadLength, payloadHash,
-                         generatorPublicKey, generationSignature, blockSignature, previousBlockHash,
-                         cumulativeDifficulty, baseTarget, nextBlockId, height, id, nonce, blockATs);
+    return Block.builder()
```

Comment:
Is this relevant for unit tests, or "just" a useful refactoring?

Reply:
I would say it is a useful refactoring because we are more flexible when creating a new Block (which can help when creating test cases which do not need every attribute to be filled). Furthermore it increases readability IMHO.

Reply:
I just wanted to show a possibility which Lombok offers. It increases readibility as you can see at first sight, which param is set to which value. You are right, that the current state in too generic and allows the creation of "invalid" block objects. 

From my point of view the existing code needs to be improved because the readability and documentation is not so good. If you provide different constructors there should be at least a JavaDoc when to use which constructor. A better way to go would be to just use a package private constructor (so that no one can access from a "higher" level) and use the factory pattern to create complex objects / separate the different types of creation. 

In the old code there were 2 constructors. Could you provide some more information for the different creation approaches? Maybe it could be also reduced to 1? Otherwise my suggestion would be to provide different factory classes / methods to create a block for the different use case. If you think that would be a good idea, i can write the needed classes/methods. If you have another idea in mind just give me an example.